### PR TITLE
add-functionality

### PR DESCRIPTION
**Pull request by AmeerQatmosh**
**Changes:**
    • adding "ShowUserList" functional component, to display a list of users.
    • adding "ShowUserProfile" functional component  to display details for each user.
    • adding functionality.
    • adding navigation.
**Fixes:**
    • show details on "Profile" screen.
**Pending TO DO:**
    • Add Image Placeholder.
    • Improve User Interface.

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -2,65 +2,120 @@ import React, { useEffect, useState } from "react";
...
+        <Stack.Screen name="Home" component={ShowUserList} />
+        <Stack.Screen name="Profile" component={ShowUserProfile} />
+      </Stack.Navigator>
+    </NavigationContainer>
+  );
+}
 
-  const [data, setData] = useState([]);
-  const [loading, setLoading] = useState(true);
+//   const [data, setData] = useState([]);
```
Comment (dtubaileh) [inline App.js]: please remove all of those comments

File: src/component/HomePage.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from "react";
+import {Text,View,StyleSheet,ScrollView,Image,TouchableOpacity,Button, Pressable,TouchableHighlight} from 'react-native';
+import { NavigationContainer } from '@react-navigation/native';
+import { createNativeStackNavigator } from '@react-navigation/native-stack';
+import ShowUserProfile from "./UserProfile";
+
+export default function ShowUserList ({ navigation, route }){
+
+  const [data, setData] = useState([]);
+  const [state, setState] = useState([]);
```
Comment (dtubaileh) [inline src/component/HomePage.js]: this state is un used, please remove it

File: src/component/UserProfile.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import { createNativeStackNavigator } from '@react-navigation/native-stack';
+import ShowUserList from "./HomePage";
+
+export default function ShowUserProfile (props){
+  return(
+    <SafeAreaView style={styles.container}>
+      <ScrollView style={styles.scrollView}>
+        <Image
+          style={styles.image}
+          source={{uri: props.route.params.item.user.avatar_url}}
```
Comment (dtubaileh) [inline src/component/UserProfile.js]: why do you need "item" in props?

File: src/component/UserProfile.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        />  
+        <Text style={{textAlign: 'center',fontSize: 20,fontFamily: 'monospace',color: 'black'}} > {props.route.params.item.user.login}</Text>
+        
+        <Text style={styles.text} > <Text style={styles.bold}>ID:</Text> {props.route.params.item.user.id}</Text>
```
Comment (dtubaileh) [inline src/component/UserProfile.js]: when you see deprecated sequential of variable, combine them it in one line

**Review by dtubaileh — COMMENTED** [review]
please solve all the comment that I had added, let me know if anything is not clear

File: App.js (hunk 1 above)
Comment (AmeerQatmosh) [reply App.js]: Done

File: src/component/HomePage.js (hunk 2 above)
Comment (AmeerQatmosh) [reply src/component/HomePage.js]: Done

File: src/component/UserProfile.js (hunk 3 above)
Comment (AmeerQatmosh) [reply src/component/UserProfile.js]: We don't need it, I just removed.

File: src/component/UserProfile.js (hunk 4 above)
Comment (AmeerQatmosh) [reply src/component/UserProfile.js]: I do not understand?

**Pull request merged**